monai/data/writers/niftiwriter.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2020 MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import nibabel as nib
+
+
+def write_nifti(data, affine, file_name, revert_canonical, dtype="float32"):
+    """Write numpy data into nifti files to disk.
+
+    Args:
+        data (numpy.ndarray): input data to write to file.
+        affine (numpy.ndarray): affine information for the data.
+        file_name (string): expected file name that saved on disk.
+        revert_canonical (bool): whether to revert canonical.
+        dtype (np.dtype, optional): convert the loaded image to this data type.
+
+    """
+    assert isinstance(data, np.ndarray), 'input data must be numpy array.'
+    if affine is None:
+        affine = np.eye(4)
+
+    if revert_canonical:
+        codes = nib.orientations.axcodes2ornt(nib.orientations.aff2axcodes(np.linalg.inv(affine)))
```

Comment:
a follow-up of https://github.com/Project-MONAI/MONAI/pull/53#discussion_r375914183 I don't fully understand this reverting, could we chat with the original author? @Nic-Ma

Reply:
@wyli This is being updated with tests in clara. I'll create a new issue to update this functionality next week.

Reply:
@wyli @ericspod submitted a pull request to this branch that fixes closest_canonical case. Basically we need to store the original affine matrix and then use that to calculate the orientation matrix and reverse values using that. See #74